### DISCUSSION TITLE
WASM Support?

### DISCUSSION THREAD

**Discussion by mauricioszabo** [jank-lang/jank#13] [Feature Requests & Ideas] [answered]
As Jank is targetting LLVM, do you think it's possible to compile to WASM?

It would be really interesting as a target for Jank, and it would open new doors to Clojure web development too.

**Comment by jeaye** [comment, marked as answer]
Hey! Thanks for your interest.

I do think this will be possible, thanks to LLVM. It's not a primary goal, just yet, but we can be mindful to make sure we don't design in a way which prevents it.

I'm not yet sure how this would affect things like the JIT compilation (since I'm not familiar with WASM's JIT capabilities), but it should be relatively straight forward for AOT builds, based on what I'm seeing in articles like this: https://evacchi.github.io/llvm/wasm/wasi/2022/04/14/compiling-llvm-ir-into-wasm.html
